spring-cloud-starter-stream-source-sftp-dataflow/README.adoc

```diff
@@ -0,0 +1,130 @@
+//tag::ref-doc[]
+= SFTP Data Flow Source
+
+This is a specialized version of the `sftp` source which always outputs a Data Flow Task Launch Request as JSON,
+compatible with the https://github.com/spring-cloud-stream-app-starters/tasklauncher-data-flow[tasklauncher-data-flow
+] sink.
+
+NOTE: A task launch request posted to the Data Flow Server API is much simpler to use than the `TaskLaunchRequest`
+supported by the original `sftp` source, designed to launch tasks using one of the provided platform specific task
+launchers. Using a platform specific task launcher makes it possible to launch tasks when a Data Flow server is not
+deployed, but requires several additional configuration parameters.
+
+This source transfers files using the SFTP protocol from the `remote` directory to the `local` directory where the application is deployed.
+
+NOTE: When running on a cloud platform such as (Cloud Foundry 2.3+ or Kubernetes), it is recommended that the local
+directory
+reference a shared mount path provided by platform volume services which can also be accessed from the launched task
+container.
+
+Optionally, if 'list-only=true', the Task Launch Request provides command line arguments to enable the launched task
+to access the remote file from the sftp server. Using a local or shared volume is the recommended approach.
+The 'list-only` option is only provided as a fallback for cases in which a shared volume is not an option.
+
+
+When configuring the `sftp.factory.known-hosts-expression` option, the root object of the evaluation is the application context, an example might be `sftp.factory.known-hosts-expression = @systemProperties['user.home'] + '/.ssh/known_hosts'`.
+
+See also https://github.com/spring-cloud-stream-app-starters/core/blob/master/common/app-starters-metadata-store-common/README.adoc[`MetadataStore`] options for possible shared persistent store configuration for the `FtpPersistentAcceptOnceFileListFilter` and `IdempotentReceiverInterceptor` used in the FTP Source.
```

Comment:
`SftpPersistentAcceptOnceFileListFilter` and `SFTP source`?

Reply:
I thought you wrote that :) It's copied from the original README

Reply:
OK. Feel free to raise an issue against me then! 😉